From 857a3a2ea2647eea68caf9a7bc5d2b10857888e0 Mon Sep 17 00:00:00 2001
From: Corentin Wallez <cwallez@chromium.org>
Date: Fri, 25 Jun 2021 17:23:26 +0200
Subject: [PATCH] CopyTextureForBrowser(): Fix macro declaration

This is fixup for the
https://dawn-review.googlesource.com/c/dawn/+/54800
where the macre was not properly declared.
Clang doesn't claims about it, but GCC does.
The errors example for GCC 9.3.0:
/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp:40:5:
error: unterminated raw string
   40 |     R"(
      |     ^
/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp:46:6:
warning: missing terminating " character
   46 |     )"
      |      ^

Bug: chromium:1217153, chromium:819294
Change-Id: I47aa2dac37d9dfa7c02532caeb3341edd22fcd07
---

diff --git a/third_party/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp b/third_party/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp
index 8dbc9aa..46da2c8 100644
--- a/third_party/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp
+++ b/third_party/dawn/src/dawn_native/CopyTextureForBrowserHelper.cpp
@@ -36,14 +36,12 @@
 // TODO(crbug.com/1221110): Remove this header macro by merging vertex and
 // fragment shaders into one shader source. Now it blocks by
 // crbug.com/dawn/947 and crbug.com/tint/915
-#define HEADER \
-    R"(
-            [[block]] struct Uniforms {
-                u_scale: vec2<f32>;
-                u_offset: vec2<f32>;
-                u_alphaOp: u32;
-            };
-    )"
+#define HEADER                              \
+    "        [[block]] struct Uniforms {\n" \
+    "            u_scale: vec2<f32>;\n"     \
+    "            u_offset: vec2<f32>;\n"    \
+    "            u_alphaOp: u32;\n"         \
+    "        };\n"
 
         static const char sCopyTextureForBrowserVertex[] = HEADER R"(
             [[binding(0), group(0)]] var<uniform> uniforms : Uniforms;
